be/src/exec/vectorized/iceberg_delete_builder.cpp

```diff
@@ -0,0 +1,169 @@
+        int64_t file_size;
+        RETURN_IF_ERROR(parquet_reader->size(&file_size));
+        if (file_size == 0) {
+            parquet_reader->close();
+            return Status::OK();
+        }
+        std::vector<SlotDescriptor*> src_slot_descriptors;
+        src_slot_descriptors.emplace_back(new SlotDescriptor(kPosDeleteFilePathField));
+        src_slot_descriptors.emplace_back(new SlotDescriptor(kPosDeletePosField));
+        auto file_reader = std::make_shared<ParquetChunkReader>(std::move(parquet_reader), src_slot_descriptors, runtime_state->timezone());
+        Status st = Status::OK();
+        std::shared_ptr<::arrow::RecordBatch> batch;
+        do {
+            st = file_reader->next_batch(&batch);
+            if (batch) {
+                ::arrow::StringArray * file_path_array = static_cast<arrow::StringArray *>(batch->column(0).get());
+                ::arrow::Int64Array * pos_array = static_cast<arrow::Int64Array *>(batch->column(1).get());
+                for (size_t row = 0; row < batch->num_rows(); row++) {
+                    if (file_path_array->Value(row).find(path) != std::string::npos) {
+                        need_skip_rowids.emplace(pos_array->Value(row));
+                    }
+                }
+            }
+        } while (st.ok() && !st.is_end_of_file());
+        return st;
+    }
+
+    Status ParquetEqualityDeleteBuilder::build(RuntimeState* runtime_state,
+                                               std::unordered_map<SlotId, std::vector<ExprContext*>> conjunct_ctxs_by_slot,
+                                               const TIcebergDeleteFile* tdelete_file) {
+
+        std::shared_ptr<RandomAccessFile> raw_file;
+        ASSIGN_OR_RETURN(raw_file, fs->new_random_access_file(tdelete_file->full_path));
+        auto parquet_file = std::make_shared<ParquetChunkFile>(raw_file, 0);
+        // equality delete file
+        auto parquet_reader = std::make_shared<ParquetReaderWrap>(std::move(parquet_file), -1, 0, tdelete_file->length);
+        int64_t file_size;
+        RETURN_IF_ERROR(parquet_reader->size(&file_size));
+        if (file_size == 0) {
+            parquet_reader->close();
+            return Status::OK();
+        }
+
+        // No need to set src slot descriptors here since we
+        // should read all the columns from equality delete file
+        std::vector<SlotDescriptor*> src_slot_descriptors;
+        auto file_reader = std::make_shared<ParquetChunkReader>(std::move(parquet_reader), src_slot_descriptors, runtime_state->timezone());
+        Status st = Status::OK();
+        std::shared_ptr<::arrow::RecordBatch> batch;
+        do {
+            st = file_reader->next_batch(&batch);
+            if (batch) {
+                for (int i = 0; i < batch->num_columns(); ++i) {
+                    auto column_name = batch->schema()->field_names()[i];
+                    for (auto& slot : materialize_slots) {
```

Comment:
what happen is a column not found in `materialize_slots`

Reply:
with https://github.com/StarRocks/starrocks/pull/12376 this should not happen